Sources/Debug/RoverSettingsViewController.swift

```diff
@@ -308,8 +310,29 @@ extension SettingsViewController: UITableViewDelegate {
 
 // MARK: UINavigationBarDelegate
 
-extension SettingsViewController: UINavigationBarDelegate {
+extension RoverSettingsViewController: UINavigationBarDelegate {
     public func position(for bar: UIBarPositioning) -> UIBarPosition {
         return .topAttached
     }
 }
+
+// MARK: UITextFieldDelegate
+
+extension RoverSettingsViewController: UITextFieldDelegate {
+    public func textFieldShouldBeginEditing(_ textField: UITextField) -> Bool {
+        return true
+    }
+    
+    public func textFieldShouldEndEditing(_ textField: UITextField) -> Bool {
+        return true
+    }
+    
+    public func textFieldDidEndEditing(_ textField: UITextField) {
+        RoverFoundation.shared!.resolve(DeviceNameManager.self)?.setDeviceName(textField.text ?? UIDevice.current.name)
+    }
+    
+    public func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+        textField.resignFirstResponder()
+        return true
+    }
+}
```

Comment:
haha the funny thing is if I were doing this I just woulda been like "F it" and used SwiftUI 😂 

but keeping with uikit is just fine, since technically target SDK till lower than 13 for now.

Reply:
Yeah, I don't see the benefit of forcing developers to add a SwiftUI dependency here.